introduction/07_questionnaires.rst

```diff
@@ -114,39 +162,130 @@ Examples
     d. an irrational number
     e. -3
     *f. -5
+    ?g. neutral option
 
     a § Rather close. Remember that you can add or subtract the same number to the both sides of the equation.
-    b § Correct!
-    c § Correct!
+    !b § If option "an integer" is not chosen, this hint will be shown.
     d § No. This equation has a nice and easy solution.
-    e § Remember that :math:`x^2 = q \leftrightarrow x = \pm \sqrt{q}`
-    f § Correct!
+
+  .. pick-any:: 10
+    :partial-points:
+
+    Checkbox questions defined with ``pick-any`` can have the option ``partial-points``. 
+    Students are then granted points also for partially correct answers. You can try it out 
+    below.
+    
+    For instance, in this case there are three correct answers, the grading goes as follows:
+    1 correct = 3 points,
+    2 correct = 6 points,
+    3 correct = 10 points.
+    
+    And for the wrong answers:
+    1 wrong option chosen = 3 points deducted, 
+    2 wrong options chosen = 6 points deducted,
+    3 wrong options chosen = 10 points deducted.
+
+    When :math:`(x + 1)^2 = 16`, what is :math:`x`?
+
+    a. 4
+    *b. an integer
+    *c. 3
+    d. an irrational number
+    e. -3
+    *f. -5
+
+Freetext questions
+------------------
 
 The ``freetext`` subdirective creates text boxes and grades based on their
 input.
 
-.. questionnaire:: questionnaire_text_demo 15
-  :title: A simple multiple-choice questionnaire
+The freetext directive also accepts the following options in addition to 
+the common question options:
+
+- ``length``: (horizontal) length for the HTML text input
```

Comment:
maybe we should clarify the units we are using. (px, pt, cm, em,  %, ...)

Reply:
Yes, you can check the units. I have a feeling that this option does not accept units and it doesn't even really do anything. Is the length saved in the exercise YAML file at all?

Reply:
No, the length is not stored in the exercise YAML file. 
![image](https://user-images.githubusercontent.com/10437475/99387704-1d562380-28dd-11eb-83f1-f4f9976a1693.png)
It seems that the length defines the attribute `size` of the input element or the number of columns when the height is defined https://github.com/apluslms/a-plus-rst-tools/blob/94523f890b488a99217049cb9e07577448ce8bda/directives/questionnaire.py#L582, However, I do not see the size attribute or the defined number of cols in the HTML code. 🤔

Reply:
Yes, the height value changes the text input into multiline textarea element, but length maybe does nothing at all at the moment.